### ISSUE TITLE
Decoding empty string fails with encoding='u8'

### ISSUE THREAD

**Issue by marningmol** [msgpack/msgpack-python#234]
`>>> msgpack.version`
`(0, 4, 8)`

`>>> msgpack.unpackb(msgpack.packb("foo", encoding='u8'), encoding='u8')`
`'foo'`

`>>> msgpack.unpackb(msgpack.packb("", encoding='u8'), encoding='u8')`

msgpack/_unpacker.pyx in msgpack._unpacker.unpackb (msgpack/_unpacker.cpp:139)()
ValueError: PyMemoryView_FromBuffer(): info->buf must not be NULL

`>>> msgpack.unpackb(msgpack.packb(""), encoding='u8')`

msgpack/_unpacker.pyx in msgpack._unpacker.unpackb (msgpack/_unpacker.cpp:139)()
ValueError: PyMemoryView_FromBuffer(): info->buf must not be NULL

Basically you can't unpack b'\xa0' if you provide encoding.

**Comment by jfolz** [comment]
Is this PyPI or git-master? I can't even find ``PyMemoryView_FromBuffer()`` in the Cython-translated code.

**Comment by methane** [comment]
I can reproduce in git-master.

BTW, 'u8' is alias of 'utf_8', and much slower than 'utf_8' or 'utf-8'.
You can use 'utf-8' for workaround.

**Comment by marningmol** [comment]
Sorry, I should have mentioned; version was the latest one I could pull off PyPI.

You're right, using 'utf-8' works. Maybe make 'u8' a true alias for 'utf-8'?

**Comment by methane** [comment]
`PyUnicode_Decode()` Python/C API has some shortcut.
"utf-8" passes the short cut. "u8" won't.
That's why behavior is different.
